### PR TITLE
feat(schema): add roles table

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Create roles table 
Add user_roles to the user's table as a jsonb type value

Add the Array type to the SchemaValue list to support the user_roles infer type as an array.

## Linear Issue Reference

LOG-1424

## Testing

add the table to the dev env
ci passed

@logto-io/eng

### CODE REVIEW

File: packages/schemas/src/db-entries/user.ts [hunk 1]
```diff
@@ -55,6 +59,7 @@ export const Users: GeneratedSchema<CreateUser> = Object.freeze({
     'passwordEncrypted',
     'passwordEncryptionMethod',
     'passwordEncryptionSalt',
+    'userRoles',
```
Comment (gao-sun) [inline packages/schemas/src/db-entries/user.ts:62]: just roles？

File: packages/schemas/tables/roles.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+create table roles (
+    id varchar(128) not null,
```
Comment (gao-sun) [inline packages/schemas/tables/roles.sql:2]: should be name and description?

File: packages/schemas/src/db-entries/user.ts (hunk 1 above)
Comment (simeng-li) [reply packages/schemas/src/db-entries/user.ts:62]: would conflict with the roles table name 🤣

Comment (simeng-li) [reply packages/schemas/src/db-entries/user.ts:62]: image

Comment (simeng-li) [reply packages/schemas/src/db-entries/user.ts:62]: feel free to shoot any suggestions

File: packages/schemas/tables/roles.sql (hunk 2 above)
Comment (simeng-li) [reply packages/schemas/tables/roles.sql:2]: k

File: packages/schemas/src/db-entries/user.ts (hunk 1 above)
Comment (simeng-li) [reply packages/schemas/src/db-entries/user.ts]: aligned offline userRoles -> roleNames

**Review by gao-sun — APPROVED** [review]

**Pull request merged**